jsapp/js/universalTable/universalTable.component.tsx

```diff
@@ -0,0 +1,325 @@
+     * values are new). The second object shows previous pagination, use it to
+     * compare what has happened :)
+     */
+    newPageInfo: PaginationState,
+    oldPageInfo: PaginationState
+  ) => void;
+  // ENDPAGINATION
+}
+
+const DEFAULT_COLUMN_SIZE = {
+  size: 200, // starting column size
+  minSize: 100, // enforced during column resizing
+  maxSize: 600, // enforced during column resizing
+};
+
+/**
+ * This is a nice wrapper for the `@tanstack/react-table`. It uses only
+ * a limited selection of all possible features, and provides consistent looks.
+ *
+ * You are responsible for passing column definitions (important!) and the data
+ * to match these definitions (obviously). When using it, you need to also pass
+ * the TS type of the data item, so it knows what to expect.
+ *
+ * It has column pinning (on column definition level, i.e. you need to tell it
+ * which columns are pinned), and column resizing (works out of the box!).
+ *
+ * It has (optional) pagination. If you pass all the required props, you can
+ * expect to get user pagination requests through the callback function named
+ * `onRequestPaginationChange`.
+ */
+export default function UniversalTable<DataItem>(
+  props: UniversalTableProps<DataItem>
+) {
+  function getCommonClassNames(column: Column<DataItem>) {
+    return cx({
+      [styles.isPinned]: Boolean(column.getIsPinned()),
+    });
+  }
+
+  const columns = props.columns.map((columnDef) => {
+    return {
+      accessorKey: columnDef.key,
+      header: () => columnDef.label,
+      cell: (cellProps: CellContext<DataItem, string>) => {
+        if (columnDef.cellFormatter) {
+          return columnDef.cellFormatter(cellProps.getValue());
+        } else {
+          return cellProps.renderValue();
+        }
+      },
+      size: columnDef.size || DEFAULT_COLUMN_SIZE.size,
+    };
+  });
+
+  // We define options as separate object to make the optional pagination truly
+  // optional.
+  const options: TableOptions<DataItem> = {
+    columns: columns,
+    data: props.data,
+    getCoreRowModel: getCoreRowModel(),
+    columnResizeMode: 'onChange',
+    //override default column sizing
+    defaultColumn: DEFAULT_COLUMN_SIZE,
+  };
+
+  options.state = {};
+
+  // Set separately to not get overriden by pagination options. This is a list
+  // of columns that are pinned to the left side.
+  const pinnedColumns = props.columns
+    .filter((col) => col.isPinned)
+    .map((col) => col.key);
+  options.state.columnPinning = {left: pinnedColumns || []};
+
+  const hasPagination = (
+    props.pageIndex !== undefined &&
+    props.pageCount !== undefined &&
+    props.pageSize !== undefined &&
+    props.pageSizes !== undefined &&
+    props.onRequestPaginationChange !== undefined
+  );
+
+  // Add pagination related options if needed
+  if (
+    hasPagination &&
+    // `hasPagination` handles everything, but we need these two for TypeScript:
+    props.pageSize !== undefined &&
```

Comment:
We could switch this to an `||` too I think,

Reply:
Same as above